source/actions3/actions/AMDetectorWaitForAcquisitionStateAction.cpp

```diff
@@ -0,0 +1,78 @@
+#include "AMDetectorWaitForAcquisitionStateAction.h"
+
+#include "beamline/AMBeamlineSupport.h"
+#include "util/AMErrorMonitor.h"
+
+AMDetectorWaitForAcquisitionStateAction::AMDetectorWaitForAcquisitionStateAction(AMDetectorWaitForAcquisitionStateActionInfo *info, AMDetector *detector, QObject *parent)
+	: AMAction3(info, parent)
+{
+	if(detector)
+		detector_ = detector;
+	else if(AMBeamlineSupport::beamlineDetectorAPI())
+		detector_ = AMBeamlineSupport::beamlineDetectorAPI()->exposedDetectorByInfo(*(info->detectorInfo()));
+	else
+		detector_ = 0; //NULL
+}
+
+
+AMDetectorWaitForAcquisitionStateAction::AMDetectorWaitForAcquisitionStateAction(const AMDetectorWaitForAcquisitionStateAction &other)
+	: AMAction3(other)
+{
+	const AMDetectorWaitForAcquisitionStateActionInfo *info = qobject_cast<const AMDetectorWaitForAcquisitionStateActionInfo*>(other.info());
+
+	//if(info)
+	//	detector_ = AMBeamline::bl()->exposedDetectorByInfo(*(detectorInitializeInfo()->detectorInfo()));
+	if(info && AMBeamlineSupport::beamlineDetectorAPI())
+		detector_ = AMBeamlineSupport::beamlineDetectorAPI()->exposedDetectorByInfo(*(detectorWaitForAcquisitionStateInfo()->detectorInfo()));
+	else
+		detector_ = 0;
+}
+
+AMDetectorWaitForAcquisitionStateAction::~AMDetectorWaitForAcquisitionStateAction()
+{
+
+}
+
+void AMDetectorWaitForAcquisitionStateAction::startImplementation()
+{
+	if(!detector_ && AMBeamlineSupport::beamlineDetectorAPI())
+		detector_ = AMBeamlineSupport::beamlineDetectorAPI()->exposedDetectorByInfo(*(detectorWaitForAcquisitionStateInfo()->detectorInfo()));
+
+	if(!detector_) {
+		AMErrorMon::alert(this,
+						  AMDETECTORINITIALIZEACTION_NO_VALID_DETECTOR,
+						  QString("There was an error initializing the detector '%1', because the detector was not found. Please report this problem to the Acquaman developers.").arg(detectorWaitForAcquisitionStateInfo()->name()));
```

Comment:
By the way, if you want github to align these continuation lines properly you can set **Align continuation lines** to **With Spaces** in:
Options >> C++ >> Code Style >> General
and
Options >> Text Editor >> Beahviour